be/src/exprs/vectorized/string_functions.cpp

```diff
@@ -23,6 +24,17 @@
 
 namespace starrocks::vectorized {
 
+#define THROW_RUNTIME_ERROR_IF_EXCEED_LIMIT(col, func)                         \
+    if (col->reach_capacity_limit()) {                                         \
+        col->reset_column();                                                   \
```

Comment:
why reset column?

Reply:
query abort because of MemLimitExceeded, then reset the column to release memory timely

Reply:
col->reset_column() can't release memory

Reply:
@trueeyu to pass ASAN DCHECK
BinaryColumn will check bytes.size() == offsets.back() under ASAN;